packages/client-node/src/result_set.ts

```diff
@@ -35,34 +38,64 @@ export class ResultSet implements BaseResultSet<Stream.Readable> {
 
     validateStreamFormat(this.format)
 
-    let decodedChunk = ''
+    let data: Buffer
+    let incompleteRow: Buffer | undefined
     const toRows = new Transform({
       transform(
         chunk: Buffer,
-        encoding: BufferEncoding,
+        _encoding: BufferEncoding,
         callback: TransformCallback
       ) {
-        decodedChunk += chunk.toString()
+        let currentDataSize
+        // The buffer was already allocated
+        // x2 chunk size should be enough for any leftover incomplete rows
```

Comment:
No, it is actually not enough. I will add ad-hoc resizing cause we can have huge strings or other data with variable length out of nowhere in some chunks.